File: 백준/Bronze/2231. 분해합/분해합.py
def find_smallest_generator(N):
    for i in range(1, N + 1):
        decomposition_sum = i + sum(map(int, str(i)))
        if decomposition_sum == N:
            return i
    return 0

N = int(input())
print(find_smallest_generator(N))

File: 백준/Bronze/2231. 분해합/분해합.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");

const N = Number(input[0]);

function divideSum(num) {
  let sum = num;
  while (num > 0) {
    sum += num % 10;
    num = Math.floor(num / 10);
  }
  return sum;
}

let found = false;

for (let i = 1; i <= N; i++) {
  if (divideSum(i) === N) {
    console.log(i);
    found = true;
    break;
  }
}

if (!found) {
  console.log(0);
}
